code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -1,7 +1,52 @@
-#define CRYOMOBS 'icons/obj/cryo_mobs.dmi'
 ///Max temperature allowed inside the cryotube, should break before reaching this heat
 #define MAX_TEMPERATURE 4000
 
+/// This is a visual helper that shows the occupant inside the cryo cell.
+/atom/movable/cryo_occupant_vis
+	icon = 'icons/obj/cryogenics.dmi'
+	// Must be tall, otherwise the filter will consider this as a 32x32 tile
+	// and will crop the head off.
+	icon_state = "mask_bg"
+	layer = ABOVE_WINDOW_LAYER + 0.01
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	var/obj/machinery/atmospherics/components/unary/cryo_cell/cryo
+	var/animation = FALSE
+	pixel_y = 22
+	appearance_flags = KEEP_TOGETHER
+
+/atom/movable/cryo_occupant_vis/Initialize(mapload, obj/machinery/atmospherics/components/unary/cryo_cell/cryo_)
+	. = ..()
+	cryo = cryo_
+	// Alpha masking
+	// It will follow this as the animation goes, but that's no problem as the "mask" icon state
+	// already accounts for this.
+	filters += filter(type = "alpha", icon = icon('icons/obj/cryogenics.dmi', "mask"), y = -22)
+
+/atom/movable/cryo_occupant_vis/Destroy()
+	cryo = null
+	return ..()
+
+/atom/movable/cryo_occupant_vis/update_icon()
+	if(!cryo)
+		return
+	if(animation)
+		if(!cryo.occupant || !cryo.on || !cryo.is_operational)
+			animate(src)
+			animation = FALSE
+		return
+
+	vis_contents.Cut()
```

Comment:
Is there need to cut and repopulate this list every time?

Reply:
Will fix